pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -105,6 +95,45 @@ type Results struct {
 	PodErrors     map[*v1.Pod]error
 }
 
+// Record sends eventing and log messages back for the results that were produced from a scheduling run
+// It also nominates nodes in the cluster state based on the scheduling run to signal to other components
+// leveraging the cluster state that a previous scheduling run that was recorded is relying on these nodes
+func (r Results) Record(ctx context.Context, recorder events.Recorder, cluster *state.Cluster) {
+	// Report failures and nominations
+	for p, err := range r.PodErrors {
+		logging.FromContext(ctx).With("pod", client.ObjectKeyFromObject(p)).Errorf("Could not schedule pod, %s", err)
+		recorder.Publish(PodFailedToScheduleEvent(p, err))
```

Comment:
Looks like now we'll be emitting these events on scheduling simulations in disruption, which doesn't really make sense since we'd be emitting these events for pods that are already scheduled. We're really trying to just reference the re-created replica, so I don't think we want this event here.

Reply:
Good callout. I was thinking that we would never call this for any pods during the disruption loop, but I'm realizing that it's possible for us to make a call to this function with pod errors for pending pods that will never actually schedule to nodes.

Reply:
> which doesn't really make sense since we'd be emitting these events for pods that are already scheduled

Why would we emit failure to schedule if we made a scheduling decision? If we are making a disruption decision, it's because we didn't get any errors for scheduling currently scheduled pods. The only errors that we would get in here would be for pending pods that didn't schedule during the last scheduling round either, so I think we are fine to fire events and logs here.

Reply:
We'd be firing events here for any mirror or daemonset pods https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/utils/pod/scheduling.go#L87-L88, which I don't think we need to.

Reply:
We don't schedule either of these pods. They aren't part of our reschedulable pod set